File: BOJ/seulah/7_week/지름길.py
N,D = map(int,input().split())
shortcut = []
for _ in range(N):
    start,end,length = map(int,input().split())
    if end-start > length:
        shortcut.append([start,end,length])

shortcut.sort()

dp = [i for i in range(D+1)]

for start,end,length in shortcut:
    for i in range(1, D+1):
        if i == end :
            dp[i] = min(dp[i],dp[start]+length)
        else:
            dp[i] = min(dp[i],dp[i-1]+1)

print(dp[D])

File: BOJ/jaehong/7_week/지름길.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "BACKJOON_001446.txt";
const input = fs.readFileSync(filePath, "utf8").trim().split(/\r?\n/);

const [N, D] = input.shift().split(" ").map(Number);

const shortcuts = [];
input.forEach((line) => {
  const [start, end, time] = line.split(" ").map(Number);
  if (start > D || end > D) return;
  if (!shortcuts[start]) {
    shortcuts[start] = [];
  }
  shortcuts[start].push([end, time]);
});

const dp = Array.from({ length: D + 1 }, (_, i) => i);

for (let i = 0; i < D; i++) {
  dp[i + 1] = Math.min(dp[i + 1], dp[i] + 1);

  if (!shortcuts[i]) continue;
  shortcuts[i].forEach(([end, time]) => {
    dp[end] = Math.min(dp[end], dp[i] + time);
  });
}

console.log(dp[D]);
